### PR TITLE
Add NoisyControlledPQC layer.

### PR DESCRIPTION
**Pull request by MichaelBroughton**
Very similar to: #549 .

### CODE REVIEW

File: scripts/import_test.py [hunk 1]
```diff
@@ -58,8 +58,12 @@ def test_imports():
     _ = tfq.layers.State
     _ = tfq.layers.SampledExpectation
 
+    # High level layers.
+    _ = tfq.layers.ControlledPQC
+
```
Comment (jaeyoo) [inline scripts/import_test.py:63]: nit: please remove this duplicated.

File: tensorflow_quantum/python/layers/high_level/noisy_controlled_pqc.py [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+# ==============================================================================
+"""Module for tfq.python.layers.high_level.noisy_controlled_pqc layer."""
+import numbers
+import numpy as np
+import tensorflow as tf
+
+import cirq
+import sympy
+from tensorflow_quantum.core.ops.noise import \
+    noisy_expectation_op, noisy_sampled_expectation_op
```
Comment (jaeyoo) [inline tensorflow_quantum/python/layers/high_level/noisy_controlled_pqc.py:23]: nit:
```
from tensorflow_quantum.core.ops.noise import noisy_expectation_op
from tensorflow_quantum.core.ops.noise import noisy_sampled_expectation_op
```

File: tensorflow_quantum/python/layers/high_level/noisy_controlled_pqc.py [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+    gradients of the above example:
+
+
+    >>> bit = cirq.GridQubit(0, 0)
+    >>> model = cirq.Circuit(
+    ...     cirq.X(bit) ** sympy.Symbol('alpha'),
+    ...     cirq.Z(bit) ** sympy.Symbol('beta'),
+    ...     cirq.depolarize(0.01)(bit)
+    ... )
+    >>> outputs = tfq.layers.ControlledPQC(
```
Comment (jaeyoo) [inline tensorflow_quantum/python/layers/high_level/noisy_controlled_pqc.py:110]: outputs = tfq.layers.`Noisy`ControlledPQC( ?

File: tensorflow_quantum/python/layers/high_level/noisy_controlled_pqc.py [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+    calculation based on measured bitstrings instead of analytic calculation:
+
+
+    >>> bit = cirq.GridQubit(0, 0)
+    >>> model = cirq.Circuit(
+    ...     cirq.X(bit) ** sympy.Symbol('alpha'),
+    ...     cirq.Z(bit) ** sympy.Symbol('beta'),
+    ...     cirq.depolarize(0.01)(bit)
+    ... )
+    >>> outputs = tfq.layers.ControlledPQC(
```
Comment (jaeyoo) [inline tensorflow_quantum/python/layers/high_level/noisy_controlled_pqc.py:78]: outputs = tfq.layers.`Noisy`ControlledPQC( ?

**Review by jaeyoo — APPROVED** [review]
LGTM with minor nits

File: scripts/import_test.py (hunk 1 above)
Comment (MichaelBroughton) [reply scripts/import_test.py]: Fixed.

File: tensorflow_quantum/python/layers/high_level/noisy_controlled_pqc.py (hunk 2 above)
Comment (MichaelBroughton) [reply tensorflow_quantum/python/layers/high_level/noisy_controlled_pqc.py]: Fixed.

File: tensorflow_quantum/python/layers/high_level/noisy_controlled_pqc.py (hunk 3 above)
Comment (MichaelBroughton) [reply tensorflow_quantum/python/layers/high_level/noisy_controlled_pqc.py]: Good catch fixed.

File: tensorflow_quantum/python/layers/high_level/noisy_controlled_pqc.py (hunk 4 above)
Comment (MichaelBroughton) [reply tensorflow_quantum/python/layers/high_level/noisy_controlled_pqc.py]: Fixed.

**Review by jaeyoo — APPROVED** [review]
LGTM

**Pull request merged**